src/vm/hints/find_element_hint.rs

```diff
@@ -165,21 +165,136 @@ pub fn find_element(
     }
 }
 
+pub fn search_sorted_lower(
+    vm: &mut VirtualMachine,
+    ids: HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    let array_ptr_addr = get_address_from_var_name("array_ptr", &ids, vm, hint_ap_tracking)?;
+    let elm_size_addr = get_address_from_var_name("elm_size", &ids, vm, hint_ap_tracking)?;
+    let n_elms_addr = get_address_from_var_name("n_elms", &ids, vm, hint_ap_tracking)?;
+    let index_addr = get_address_from_var_name("index", &ids, vm, hint_ap_tracking)?;
+    let key_addr = get_address_from_var_name("key", &ids, vm, hint_ap_tracking)?;
+
+    match (
+        vm.memory.get(&array_ptr_addr),
+        vm.memory.get(&elm_size_addr),
+        vm.memory.get(&n_elms_addr),
+        vm.memory.get(&index_addr),
+        vm.memory.get(&key_addr),
+    ) {
+        (
+            Ok(Some(maybe_rel_array_ptr)),
+            Ok(Some(maybe_rel_elm_size)),
+            Ok(Some(maybe_rel_n_elms)),
+            Ok(_),
+            Ok(Some(maybe_rel_key)),
+        ) => {
+            let _ = vm
+                .builtin_runners
+                .iter()
+                .find(|(name, _)| name.as_str() == "range_check")
+                .ok_or(VirtualMachineError::NoRangeCheckBuiltin)?
+                .1
+                .as_any()
+                .downcast_ref::<RangeCheckBuiltinRunner>()
+                .ok_or(VirtualMachineError::NoRangeCheckBuiltin)?;
+
+            let elm_size = if let MaybeRelocatable::Int(ref elm_size) = maybe_rel_elm_size {
+                elm_size
+            } else {
+                return Err(VirtualMachineError::ExpectedInteger(
+                    maybe_rel_elm_size.clone(),
+                ));
+            };
+
+            if !elm_size.is_positive() {
+                return Err(VirtualMachineError::ValueOutOfRange(elm_size.clone()));
+            }
+
+            let n_elms = if let MaybeRelocatable::Int(ref n_elms) = maybe_rel_n_elms {
+                n_elms
+            } else {
+                return Err(VirtualMachineError::ExpectedInteger(
+                    maybe_rel_n_elms.clone(),
+                ));
+            };
+
+            if n_elms.is_negative() {
+                return Err(VirtualMachineError::ValueOutOfRange(n_elms.clone()));
+            }
+
+            if let Some(variables) = vm.exec_scopes.get_local_variables() {
+                if let Some(PyValueType::BigInt(find_element_max_size)) =
+                    variables.get("find_element_max_size")
+                {
+                    if n_elms > find_element_max_size {
+                        return Err(VirtualMachineError::FindElemMaxSize(
+                            find_element_max_size.clone(),
+                            n_elms.clone(),
+                        ));
+                    }
+                }
+            }
```

Comment:
There's also a new helper for getting ints from the scope IIRC.

Reply:
resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/305/commits/c3d02a7a40339b3dfb695aaea764d78c5fe4f589